rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,829 @@
+                    registry.latest_version(),
+                )
+                .unwrap();
+
+            let decoded = NodeRecord::decode(node.value.as_slice()).unwrap();
+            assert_eq!(node.version, version_before + 1);
+            assert_eq!(decoded.node_operator_id, operator(2).as_slice())
+        }
+
+        for (operator, allowance) in &[(operator(1), 13), (operator(2), 7)] {
+            let operator_record = registry
+                .get(
+                    make_node_operator_record_key(*operator).as_bytes(),
+                    registry.latest_version(),
+                )
+                .unwrap();
+
+            let decoded = NodeOperatorRecord::decode(operator_record.value.as_slice()).unwrap();
+            assert_eq!(operator_record.version, version_before + 1);
+            assert_eq!(decoded.node_allowance, *allowance);
+        }
+    }
+
+    #[test]
+    fn allow_duplicate_correct_records() {
+        let mut registry = Registry::new();
+
+        let first_node = generate_valid_node_id();
+        registry.apply_mutations_for_test(
+            vec![
+                dc_mutation("dc1"),
+                operator_mutation(operator(1), provider(1), 10, "dc1"),
+                operator_mutation(operator(2), provider(1), 10, "dc1"),
+                node_mutation(first_node.node_id(), operator(1)),
+            ]
+            .into_iter()
+            // These are the mutations required to have a compliant registry
+            .chain(get_mutations_to_achieve_invariancy(&[&first_node]))
+            .collect(),
+        );
+
+        let version_before = registry.latest_version();
+        registry
+            .do_replace_operator_(
+                payload(
+                    operator(1),
+                    operator(2),
+                    &[first_node.node_id(), first_node.node_id()],
+                ),
+                caller(1),
+            )
+            .assert_ok();
+
+        assert!(
+            registry.latest_version() == version_before + 1,
+            "Expected registry version to increase. Before execution: {}, after execution: {}",
+            version_before,
+            registry.latest_version()
+        );
+        for (operator, allowance) in &[(operator(1), 11), (operator(2), 9)] {
+            let operator_record = registry
+                .get(
+                    make_node_operator_record_key(*operator).as_bytes(),
+                    registry.latest_version(),
+                )
+                .unwrap();
+
+            let decoded = NodeOperatorRecord::decode(operator_record.value.as_slice()).unwrap();
+            assert_eq!(operator_record.version, version_before + 1);
+            assert_eq!(decoded.node_allowance, *allowance);
+        }
+    }
+
+    // Functions below are used to create an invariant
+    // registry. They are not important for the tests above
+    fn get_mutations_to_achieve_invariancy(
```

Comment:
We have an "invariant_compliant_registry" function that you can start with in test_helpers.rs

Reply:
I can start, but I would have to remove the node record it adds as it doesn't comply with the requirements for all the public key's the node has to have. So for me it was actually harder to start from there and remove all the things and redo it back. Maybe I made some mistakes but I do feel like another version of `invariant_compliant_registry` function should be added to `test_helpers` where the checks would be asserted upon creation.

Reply:
Removed all of the functions below since I managed to refactor tests to use existing pieces together.